consensus/src/test/kotlin/maru/consensus/validation/BlockValidatorImplTest.kt

```diff
@@ -0,0 +1,257 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus.validation
+
+import maru.consensus.ValidatorProvider
+import maru.core.BeaconBlock
+import maru.core.ExecutionPayload
+import maru.core.HashUtil
+import maru.core.Seal
+import maru.core.SealedBeaconBlock
+import maru.core.Validator
+import maru.core.ext.DataGenerators
+import maru.database.Database
+import maru.executionlayer.client.ExecutionLayerClient
+import maru.serialization.rlp.bodyRoot
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argThat
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadStatusV1
+import tech.pegasys.teku.ethereum.executionclient.schema.Response
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import tech.pegasys.teku.spec.executionlayer.ExecutionPayloadStatus
+
+class BlockValidatorImplTest {
+  private val validPrevBlockBody = DataGenerators.randomBeaconBlockBody()
+  private val validPrevBlockHeader =
+    DataGenerators.randomBeaconBlockHeader(9u).copy(
+      bodyRoot = HashUtil.bodyRoot(validPrevBlockBody),
+    )
+  private val validPrevBlock = BeaconBlock(validPrevBlockHeader, validPrevBlockBody)
+  private val validBlockBody = DataGenerators.randomBeaconBlockBody()
+  private val validBlockHeader =
+    DataGenerators.randomBeaconBlockHeader(validPrevBlockHeader.number + 1u).copy(
+      parentRoot = validPrevBlockHeader.hash,
+      timestamp = validPrevBlockHeader.timestamp + 1u,
+      bodyRoot = HashUtil.bodyRoot(validBlockBody),
+    )
+  private val validBlock = BeaconBlock(validBlockHeader, validBlockBody)
+
+  private val validatorProvider =
+    mock<ValidatorProvider> {
+      on { getProposerForBlock(any()) }.thenReturn(validBlockHeader.proposer)
+      on { getValidatorsForBlock(any()) }
+        .thenReturn((0 until validBlockBody.prevCommitSeals.size).map { DataGenerators.randomValidator() }.toSet())
+    }
+  private val sealVerifier =
+    mock<SealVerifier> {
+      on { verifySeal(any(), any()) } doReturn (Result.failure(RuntimeException("Invalid seal")))
+      on {
+        verifySeal(
+          argThat { seal: Seal ->
+            validBlockBody.prevCommitSeals.contains(seal)
+          },
+          eq(validPrevBlockHeader),
+        )
+      } doReturn (Result.success(true))
+    }
+
+  private val executionLayerClient =
```

Comment:
I suggest returning `SafeFuture.completedFuture(Response(mockPayload))` here by default and make if overridable for the negative tests. It would make the negative tests more readable and it would be easier to follow the expected failure, without having to jump across the file.